### PR TITLE
[FSSDK-9494] Bug bash & fixes

### PR DESCRIPTION
**Pull request by mikechu-optimizely**
## Summary
- Add manual testing (bug bash) resources & Vite app
- Fix some of the resulting discovered issues

💡 Note: There are a lot of files, but mostly need review of things outside the /bug-bash/ dir.

## Test plan

- All automated testing should continue to pass.

## Issues
- FSSDK-9494

### CODE REVIEW

**Review by raju-opti — CHANGES_REQUESTED** [review]

File: src/client.ts [hunk 1]
```diff
@@ -243,8 +243,7 @@ class OptimizelyReactSDKClient implements ReactSDKClient {
         this.isClientReady = true;
       });
 
-      this.dataReadyPromise = Promise.all([this.userPromise, this._client!.onReady()]).then(res => {
```
[the remark this answers is not in the document]
Comment (mikechu-optimizely) [reply src/client.ts:246]: Thoughts on this solution @raju-opti 

Tests fail now if 
``` 
return {
            success: bothSuccessful, // is false
```

I'll need to trace down the logic and update the code or tests.  

Perhaps this more in-depth dive needs to go into a new Jira?:

[the remark this answers is not in the document]
Comment (raju-opti) [reply src/client.ts:246]: Yeah, maybe better to do this under another ticket. approving this PR

**Review by raju-opti — CHANGES_REQUESTED** [review]

File: src/client.ts [hunk 2]
```diff
@@ -239,23 +239,24 @@ class OptimizelyReactSDKClient implements ReactSDKClient {
...
 
     if (this._client) {
-      this._client.onReady().then(() => {
-        this.isClientReady = true;
-      });
-
-      this.dataReadyPromise = Promise.all([this.userPromise, this._client!.onReady()]).then(res => {
+      this.dataReadyPromise = Promise.all([this.userPromise, this._client.onReady()]).then(
+        ([userResult, clientResult]) => {
+          this.isClientReady = true;
```
[the remark this answers is not in the document]
Comment (mikechu-optimizely) [reply src/client.ts]: You're correct.  😬

**Review by raju-opti — APPROVED** [review]
lgtm

**Pull request merged**